renku/core/models/workflow/composite_plan.py

```diff
@@ -45,38 +45,31 @@ class CompositePlan(AbstractPlan):
     def __init__(
         self,
         *,
+        derived_from: str = None,
         description: str = None,
         id: str,
         invalidated_at: datetime = None,
         keywords: List[str] = None,
+        links: List[ParameterLink] = None,
+        mappings: List[ParameterMapping] = None,
         name: str,
-        derived_from: str = None,
-        plans: List[Union["CompositePlan", Plan]] = None,
+        plans: List[Plan] = None,
```

Comment:
Should it be `AbstractPlan` ? A `CompositePlan` created by a user could contain a `CompositePlan`

Reply:
👍
I made it as it was before (`Union["CompositePlan", Plan]`) which should also help with ide type linter.